src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,247 @@
+import React, {
+  useMemo, useState, useEffect,
+} from 'react';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { NewPerson } from '../types';
+import { FilterType, SortType, PropName } from '../types/enum';
+
+interface Props {
+  people: NewPerson[],
+  loading: boolean,
+  sexFilter: FilterType,
+  sexFilterHandler: (value: FilterType) => void,
+  setQuery: (value: string) => void,
+  query: string,
+  deleteQuery: () => void,
+  isError: boolean,
+  fetchPeople: () => Promise<void>;
+}
+
+export const PeoplePage: React.FC<Props> = ({
+  people,
+  loading,
+  sexFilter,
+  sexFilterHandler,
+  setQuery,
+  query,
+  deleteQuery,
+  isError,
+  fetchPeople,
+}) => {
+  const [sortField, setsortField] = useState<string | null>(null);
+  const [sortOrder, setSortOrder] = useState(SortType.original);
+  const [clickCount, setClickCount] = useState(0);
+  const [selectedCentury, setSelectedCentury]
+  = useState<number[]>([]);
+  const [isFiltered, setIsfiltered] = useState(false);
+
+  const allCenturySelection = () => {
+    setSelectedCentury([]);
+  };
+
+  const handleSort = (column: string) => {
+    setsortField(column);
+    setClickCount(clickCount + 1);
+    if (sortField === column) {
+      if (clickCount === 1) {
+        setSortOrder(SortType.asc);
+      }
+
+      if (clickCount === 2) {
+        setSortOrder(SortType.desc);
+      }
+    }
+
+    if (clickCount === 3) {
+      setsortField(column);
+      setClickCount(1);
+      setSortOrder(SortType.original);
+    }
+  };
+
+  const resetEveryThing = () => {
+    setSelectedCentury([]);
+    setQuery('');
+    sexFilterHandler(FilterType.All);
+  };
+
+  const setCurrentQuery = (currentQuery: string) => {
+    setQuery(currentQuery);
+  };
+
+  const handleCenturySelection = (century: number) => {
+    if (selectedCentury.includes(century) && selectedCentury.length === 5) {
```

Comment:
Move this value `selectedCentury.includes(century)` to a variable, you use it not once in this function